easybuild/easyblocks/x/xmipp.py

```diff
@@ -25,174 +25,203 @@
+        }
+
+        deps = [
+            # dep name, is required dep?
+            ('HDF5', None, True),
+            ('SQLite', None, True),
+            ('LibTIFF', None, True),
+            ('libjpeg-turbo', None, True),
+            ('Java', 'JAVA_HOME', True),
+            ('MATLAB', 'MATLAB_DIR', False),
+        ]
+
+        regex_subs = []
+
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            params.update({'CUDA_BIN': os.path.join(cuda_root, 'bin')})
+            params.update({'CUDA_LIB': os.path.join(cuda_root, 'lib64')})
+            params.update({'NVCC': os.environ['CUDA_CXX']})
+            # Their default for NVCC is to use g++-5, fix that
+            nvcc_flags = '--x cu -D_FORCE_INLINES -Xcompiler -fPIC '
+            nvcc_flags = nvcc_flags + '-Wno-deprecated-gpu-targets -std=c++11'
+            params.update({'NVCC_CXXFLAGS': nvcc_flags})
+            self.use_cuda = True
+
+        for dep in ['CUDA', 'MATLAB']:
+            use_dep = bool(get_software_root(dep))
+            params.update({dep: use_dep})
+
+        if get_software_root('OpenCV'):
+            params.update({'OPENCV': True})
+            if self.use_cuda:
+                params.update({'OPENCVSUPPORTSCUDA': True})
+            if LooseVersion(get_software_version('OpenCV')) >= LooseVersion('3'):
+                params.update({'OPENCV3': True})
+
+        missing_deps = []
+        for dep, var, required in deps:
+            root = get_software_root(dep)
+            if root:
+                if var:
+                    params.update({var: root})
+            elif required:
+                missing_deps.append(dep)
+
+        if missing_deps:
+            raise EasyBuildError("One or more required dependencies not available: %s", ', '.join(missing_deps))
+
+        # Here we know that Java exists since it is a required dependency
+        jnipath = os.path.join(get_software_root('Java'), 'include')
+        jnipath = os.pathsep.join([jnipath, os.path.join(jnipath, 'linux')])
+        params.update({'JNI_CPPPATH': jnipath})
+
+        for (key, val) in params.items():
+            regex_subs.extend([(r'^%s\s*=.*$' % key, r'%s = %s' % (key, val))])
+
+        apply_regex_substitutions(self.cfgfile, regex_subs)
+
+        self.setup_xmipp_env()
+
+        py_ver = get_software_version('Python')
+        py_maj_ver = LooseVersion(py_ver).version[0]
+        py_min_ver = LooseVersion(py_ver).version[1]
+        pyver = 'python%s.%s' % (py_maj_ver, py_min_ver)
+        pyincpath = os.path.join(get_software_root('Python'), 'include', pyver)
+        # Temp workaround for missing include/pythonx.y in CPATH
+        env.setvar('CPATH', os.pathsep.join([os.environ['CPATH'], pyincpath]))
+
+        super(EB_Xmipp, self).configure_step()
 
     def build_step(self):
-        """No custom build step (see install step)."""
-        pass
+        """Custom build step for Xmipp."""
+
+        # First build cuFFTAdvisor
```

Comment:
Is there a specific reason to build this first?

Reply:
It's required to be built first. Fails if not. Enhanced the comment a bit.